File: jambeth/jAmbeth.Service/src/de/osthus/ambeth/merge/model/IObjRef.java
package de.osthus.ambeth.merge.model;

import de.osthus.ambeth.annotation.XmlType;

/**
 * Contains reference information about an Object in the cache or not loaded yet. By use of these information, a specific Object is uniquely defined and can be
 * loaded from the cache.
 * 
 * @see de.osthus.ambeth.cache.ICache#getObject(IObjRef, java.util.Set<de.osthus.ambeth.cache.CacheDirective>);
 */
@XmlType
public interface IObjRef
{
	byte getIdNameIndex();

	void setIdNameIndex(byte idNameIndex);

	Object getId();

	void setId(Object id);

	Object getVersion();

	void setVersion(Object version);

	Class<?> getRealType();

	void setRealType(Class<?> realType);
}

File: ambeth/Ambeth.Service/ambeth/merge/model/IObjRef.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using De.Osthus.Ambeth.Util;
using De.Osthus.Ambeth.Annotation;

namespace De.Osthus.Ambeth.Merge.Model
{
    [XmlType(Name = "IObjRef", Namespace = "http://schemas.osthus.de/Ambeth")]
    public interface IObjRef : IPrintable
    {
        sbyte IdNameIndex { get; set; }

        Object Id { get; set; }

        Object Version { get; set; }
        
        Type RealType { get; set; }
    }
}
